### ISSUE TITLE
Mac checksum failures

### ISSUE THREAD

**Issue by mattheworiordan** [ably/docs#303]
There are a multitude of reasons why these fail, but we should help users debug these issues:

* [x] Ensure a URL to the help article is included in the error message (covered in https://github.com/ably/wiki/issues/142)
* [ ] Add more debugging hints to https://support.ably.io/solution/articles/3000067415-40101-request-mac-does-not-match
* [ ] Consider adding a token generation tool to the website for testing perhaps?

@SimonWoolf thoughts?

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/EDU-1787) by [Unito](https://www.unito.io)

**Comment by paddybyers** [comment]
Do we know that people are doing home-grown token request generation? If so, do we know why?

**Comment by mattheworiordan** [comment]
> Do we know that people are doing home-grown token request generation? 

No, I don't think they are. But we've had a few people with Mac issues now... This last one turned out to be [an Ably IOS bug](https://github.com/ably/ably-ios/issues/618) though, but I believe @SimonWoolf was talking to a customer yesterday who had a different issue.

**Comment by SimonWoolf** [comment]
Was talking to two customers yesterday with token request issues --
- [this guy](https://app.intercom.io/a/apps/ua39m1ld/respond/inbox/all/conversations/10861495852) why had the iOS bug (I was just waiting for him to confirm that the workaround of specifying a ttl when creating the token request fixed it before filing an ios bug, but then Matt beat me to it :) ) and
- [this guy](https://app.intercom.io/a/apps/ua39m1ld/respond/inbox/all/conversations/10861930672) who somehow has the ttl in his token request disappear at some point between creating it with php and it being received by ably-js. This one I'm puzzled by -- afaics there's [no room for it to go wrong](https://github.com/ably/ably-php/blob/master/src/Auth.php#L338) on the php side, or the ably-js side, and it's hard to imagine a json serializer removing a `ttl` value from an object. Have filed https://github.com/ably/ably-php/issues/46 to keep track, but am not sure it's an ably-php issue.

> Consider adding a token generation tool to the website for testing perhaps?

if we can fix the causes other than the secret being wrong, that'd probably be overkill. Agree with the other two.
